utils/check_all_upgrades.sh

```
@@ -182,21 +181,30 @@ compatible_upgrade()
   RUNTESTFLAGS="-v --extension --upgrade-path=${UPGRADE_PATH} ${USERTESTFLAGS}" \
-  unbuffer ${MAKE} -C ${REGDIR} check "TESTS=${SRCDIR}/regress/core/regress.sql" ${MAKE_ARGS} > ${TMPDIR}/log 2>&1
+  run_check > "${TMPDIR}/log" 2>&1
   if test $? = 0; then
     echo "FAIL: ${test_label} did not error out:"
-    tail ${TMPDIR}/log
+    tail "${TMPDIR}/log"
     failed
   else
-    ERR=$( grep 'ERROR:.*Downgrade .* forbidden' ${TMPDIR}/log )
+    ERR=$( grep 'ERROR:.*Downgrade .* forbidden' "${TMPDIR}/log" )
     test -n "$ERR" && {
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)